app/scripts/modules/cluster/reports/services/ngmClusterLocations.js

```diff
@@ -802,9 +830,27 @@ angular.module( 'ngmReportHub' )
         // set site_type && site_implementation
         ngmClusterLocations.setSiteTypeAndImplementationSelect(project);
         // filter list admin1,admin2,admin3,admin4,admin5
+        // angular.forEach(locations, function (location, $index) {
+        //   ngmClusterLocations.filterLocations(project, $index, location);
+        // });
+
+        var locationAdmin1pcodeTag = ''
         angular.forEach(locations, function (location, $index) {
-          ngmClusterLocations.filterLocations(project, $index, location);
+          $http({
+            method: 'GET',
+            url: ngmAuth.LOCATION + '/api/list/getAdminSites?admin0pcode='
+              + project.definition.admin0pcode
+              + '&admin1pcode=' + location.admin1pcode
+          }).then(function (result) {
+            if (locationAdmin1pcodeTag !== location.admin1pcode) {
```

Comment:
that will make a request on each location, on the same admin1 that will return 304, 
what will happen if locationAdmin1pcodeTag changes from A to B and then to A ( on refactor-location-handling locationAdmin1pcodeTag that is handled with array )
what if having a function on form widget that handles for locations unique admin sites requests if missing ( or whatever admin1,2,3 or site type by config, if taking into consideration configurable admin fetch service )

Reply:
yes  i will change it like on refactor-location-handling,